src/coreclr/vm/interoputil.cpp

```diff
@@ -2265,14 +2265,21 @@ OLE_COLOR ConvertSystemColorToOleColor(OBJECTREF *pSrcObj)
     }
     CONTRACTL_END;
 
+    OLE_COLOR result;
+
+    OBJECTREF srcObj = *pSrcObj;
+    GCPROTECT_BEGIN(srcObj);
     // Retrieve the method desc to use for the current AD.
     MethodDesc *pSystemColorToOleColorMD =
         GetAppDomain()->GetLoaderAllocator()->GetMarshalingData()->GetOleColorMarshalingInfo()->GetSystemColorToOleColorMD();
     MethodDescCallSite systemColorToOleColor(pSystemColorToOleColorMD);
 
     // Set up the args and call the method.
-    SYSTEMCOLOR *pSrcSysColor = (SYSTEMCOLOR *)(*pSrcObj)->UnBox();
-    return systemColorToOleColor.CallWithValueTypes_RetOleColor((const ARG_SLOT *)&pSrcSysColor);
+    SYSTEMCOLOR *pSrcSysColor = (SYSTEMCOLOR *)srcObj->UnBox();
+    result = systemColorToOleColor.CallWithValueTypes_RetOleColor((const ARG_SLOT *)&pSrcSysColor);
+    GCPROTECT_END();
+
+    return result;
```

Comment:
It looks like a long standing bug. `GetOleColorMarshalingInfo` has `GC_TRIGGERS`, so the `OBJECTREF` may be invalidated.

Reply:
> `GetOleColorMarshalingInfo` has `GC_TRIGGERS`, so the `OBJECTREF` may be invalidated.

Just because a method has `GC_TRIGGERS`, doesn't mean that `OBJECTREF` will be invalidated. If the method is called within cooperative mode, this is fine. Can you elaborate on the issue you are hitting a bit more?

Reply:
This `GCPROTECT` region should not be needed here. The typically pattern here is to name the input argument with the `PROTECTED` suffix - this is common throughout the codebase. The current code is definitely ugly, but I'm not sure the update here is any clearer. I would remove these changes unless they are needed for some reason.

Reply:
I've seen a CI run that reports broken OBJECTREF in this method. The srcObj doesn't seem protected anywhere.

Reply:
It looks like the bug is in `Variant_ConvertSystemColorToOleColor()`, that should have a `GCPROTECT` region. This one can be removed.

Reply:
The `DispParamOleColorMarshaler` is coming from managed invocation. It is correct.